src/coverage-system/coverageservice.ts

```diff
@@ -152,24 +150,40 @@ export class CoverageService {
         this.outputChannel.appendLine(outputMessage);
 
         this.coverageWatcher = workspace.createFileSystemWatcher(blobPattern);
-        this.coverageWatcher.onDidChange(this.loadCacheAndRender.bind(this));
-        this.coverageWatcher.onDidCreate(this.loadCacheAndRender.bind(this));
-        this.coverageWatcher.onDidDelete(this.loadCacheAndRender.bind(this));
+        this.coverageWatcher.onDidChange(this.loadCacheAndProcess.bind(this));
+        this.coverageWatcher.onDidCreate(this.loadCacheAndProcess.bind(this));
+        this.coverageWatcher.onDidDelete(this.loadCacheAndProcess.bind(this));
     }
 
-    private handleEditorEvents(textEditors: TextEditor[]) {
+    private setStatusBarCoverage(sections: Map<string, Section>, textEditor: TextEditor | undefined ) {
+        try {
+            if (!textEditor) {
+                return this.statusBar.setCoverage(undefined);
+            }
+            const [fileCoverage] = this.sectionFinder.findSectionsForEditor(textEditor, sections);
+            const covered = fileCoverage?.lines?.hit;
+            const total = fileCoverage?.lines?.found;
```

Comment:
[Q] Is this the right data / calculation? I was a little unsure of it.

Reply:
Yah based on my understanding the spec here https://github.com/mitchhentges/lcov-rs/wiki/File-format#lcov-info-file-output-of-lcov-rs and the discussion information I dup up here https://github.com/ryanluker/vscode-coverage-gutters/discussions/313 I think this is correct hah!